modyn/selector/internal/selector_strategies/gradnorm_downsampling_strategy.py

```diff
@@ -6,4 +6,13 @@ def get_downsampling_strategy(self) -> str:
         return "RemoteGradNormDownsampling"
 
     def get_downsampling_params(self) -> dict:
-        return {"downsampled_batch_size": self.downsampled_batch_size}
+        if self.sample_before_batch:
+            params = {
+                "downsampled_batch_ratio": self.downsampled_batch_ratio,
+                "downsampling_period": self.downsampling_period,
+                "maximum_keys_in_memory": self._maximum_keys_in_memory,
+            }
+        else:
+            params = {"downsampled_batch_size": self.downsampled_batch_size}
```

Comment:
why do we always have to do this if else statement? from pure code style, maybe there is a better solution if the params always follow the same pattern in all downsampling strategies. maybe we can have a small config object or sth. aditionally, this is repeated over all strategies so we want to refactor this.

but in any case, I am not sure why it is required that way. why does batch then sample not need maximum keys in memory / couldn't we give unified information in params and just ignore them in case we do not need it?

Reply:
The idea was to pass only the strictly necessary information (e.g. batch-then-sample does not require maximum_keys_in_memory because it works batch by batch and has no memory problems). But probably the best solution is to create a general config in AbstractDownsampleStrategy and then add any extra parameters in the specific classes